src/net/udpsock.go

```diff
@@ -53,3 +63,204 @@ func (a *UDPAddr) opAddr() Addr {
 	}
 	return a
 }
+
+// ResolveUDPAddr returns an address of UDP end point.
+//
+// The network must be a UDP network name.
+//
+// If the host in the address parameter is not a literal IP address or
+// the port is not a literal port number, ResolveUDPAddr resolves the
+// address to an address of UDP end point.
+// Otherwise, it parses the address as a pair of literal IP address
+// and port number.
+// The address parameter can use a host name, but this is not
+// recommended, because it will return at most one of the host name's
+// IP addresses.
+//
+// See func Dial for a description of the network and address
+// parameters.
+func ResolveUDPAddr(network, address string) (*UDPAddr, error) {
+
+	switch network {
+	case "udp", "udp4":
+	default:
+		return nil, fmt.Errorf("Network '%s' not supported", network)
+	}
+
+	// TINYGO: Use netdev resolver
+
+	host, sport, err := SplitHostPort(address)
+	if err != nil {
+		return nil, err
+	}
+
+	port, err := strconv.Atoi(sport)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing port '%s' in address: %s",
+			sport, err)
+	}
+
+	if host == "" {
+		return &UDPAddr{Port: port}, nil
+	}
+
+	ip, err := netdev.GetHostByName(host)
+	if err != nil {
+		return nil, fmt.Errorf("Lookup of host name '%s' failed: %s", host, err)
+	}
+
+	return &UDPAddr{IP: ip, Port: port}, nil
+}
+
+// UDPConn is the implementation of the Conn and PacketConn interfaces
+// for UDP network connections.
+type UDPConn struct {
+	fd            int
+	laddr         *UDPAddr
+	raddr         *UDPAddr
+	readDeadline  time.Time
+	writeDeadline time.Time
+}
+
+// Use IANA RFC 6335 port range 49152–65535 for ephemeral (dynamic) ports
+var eport = int32(49151)
+
+func ephemeralPort() int {
+	// TODO: this is racy, if concurrent DialUDPs; use atomic?
```

Comment:
Should be fine, tinygo is concurrent but not multithreaded and there is no reason the compiler would insert a goroutine yield in this function, I'd guess.

Reply:
I removed the comment, thanks.